src/main/java/io/codelex/flightplanner/service/FlightDBService.java

```diff
@@ -0,0 +1,93 @@
+package io.codelex.flightplanner.service;
+
+import io.codelex.flightplanner.models.Airport;
+import io.codelex.flightplanner.models.Flight;
+import io.codelex.flightplanner.repository.AirportDBRepository;
+import io.codelex.flightplanner.repository.FlightDBRepository;
+import io.codelex.flightplanner.requests.AddFlightRequest;
+import io.codelex.flightplanner.requests.SearchFlightsRequest;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@ConditionalOnProperty(prefix = "flight-planner", name = "storage-type", havingValue = "database")
+public class FlightDBService implements FlightService {
+    private final FlightDBRepository flightRepository;
+    private final AirportDBRepository airportRepository;
+
+
+
+    public FlightDBService(FlightDBRepository flightRepository,
+                           AirportDBRepository airportRepository) {
+        this.flightRepository = flightRepository;
+        this.airportRepository = airportRepository;
+    }
+
+    synchronized public Flight addFlight(AddFlightRequest request) {
+        Flight newFlight = new Flight(request);
+        newFlight.setFrom(addIfAirportIfExists(newFlight.getFrom()));
+        newFlight.setTo(addIfAirportIfExists(newFlight.getTo()));
+        checkRequest(newFlight);
+        return flightRepository.save(newFlight);
+    }
+    synchronized private Airport addIfAirportIfExists(Airport airport) {
+        Optional<Airport> existingAirport = airportRepository.findById(airport.getAirport());
+        return existingAirport.orElseGet(() -> airportRepository.save(airport));
+    }
+
+
+
+    public boolean checkIfAlreadyInList(Flight flight) {
+        return flightRepository.findAll()
```

Comment:
Vot šitā nevajadzētu darīt:
flightRepository.findAll()
Ja DB būtu 10 000 ieraksti, tad viņi visu tiktu ielasīti atmiņā un tas var vienā brīdī nokārt sistēmu. Bet ir iespējams uzreiz repo meklēt pēc konkrētiem kritērijiem, vai rakstīt custom meklēšanu, kas notiek DB pusē.
Izlasi šito un arī zemāk ir manual custom queries:
https://www.baeldung.com/the-persistence-layer-with-spring-data-jpa#1-automatic-custom-queries

Tas pats arī findRequested un searchAirport